.github/workflows/ci.yml

```diff
@@ -9,16 +9,59 @@ on:
     - cron: '17 8 * * *'
 
 jobs:
-  industrial_ci:
+  ci_binary:
+    name: Noetic binary job
     runs-on: ubuntu-latest
     strategy:
       matrix:
         env:
           - {ROS_DISTRO: noetic, ROS_REPO: main}
           - {ROS_DISTRO: noetic, ROS_REPO: testing}
-          - {ROS_DISTRO: noetic, UPSTREAM_WORKSPACE: "https://raw.githubusercontent.com/ros-controls/ros_control/$ROS_DISTRO-devel/ros_control.rosinstall -ros_controllers"}
+          - {ROS_DISTRO: noetic, UPSTREAM_WORKSPACE: 'https://raw.githubusercontent.com/ros-controls/ros_control/$ROS_DISTRO-devel/ros_control.rosinstall -ros_controllers'}
     steps:
       - run: sudo apt-get update -qq && sudo apt-get upgrade
       - uses: actions/checkout@v1
       - uses: ros-industrial/industrial_ci@master
         env: ${{matrix.env}}
+
+  ci_source:
+    name: Noetic source job
+    runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: false
+    steps:
+      - uses: ros-tooling/setup-ros@v0.1
+        with:
+          required-ros-distributions: noetic
+      - uses: ros-tooling/action-ros-ci@v0.1
+        with:
+          target-ros1-distro: noetic
+          # build all packages listed in the meta package
+          package-name:
+            ackermann_steering_controller
+            diff_drive_controller
+            effort_controllers
+            force_torque_sensor_controller
+            forward_command_controller
+            four_wheel_steering_controller
+            gripper_action_controller
+            imu_sensor_controller
+            joint_state_controller
+            joint_trajectory_controller
+            position_controllers
+            ros_controllers
+            rqt_joint_trajectory_controller
+            velocity_controllers
+          vcs-repo-file-url: |
+            https://raw.githubusercontent.com/ros-controls/ros_control/noetic-devel/ros_control.rosinstall
```

Comment:
Unlike in the industrial_ci `UPSTREAM_WORKSPACE`, we don't need to exclude ros_controllers itself, since `action-ros-ci` does this automatically (See [here](https://github.com/ros-tooling/action-ros-ci/blob/2d437cb171c1b95b338c22ff67a4e8a2b87394a5/src/action-ros-ci.ts#L208)).

Unfortunately, we can't use the `$ROS_DISTRO` macro here since that's provided by industrial_ci, so instead we just specify the `noetic-devel` branch directly. No big deal.